src/include/cart/api.h

```diff
@@ -694,162 +692,142 @@ crt_ep_abort(crt_endpoint_t *ep);
+			    if (unlikely(e_ptr == NULL)) {                                         \
+				    rc = -DER_NOMEM;                                               \
+				    goto out;                                                      \
+			    }                                                                      \
+			    *e_ptrp = e_ptr;                                                       \
+		    }                                                                              \
+		    /* process the elements of array */                                            \
+		    for (i = 0; i < count; i++) {                                                  \
+			    rc = CRT_GEN_GET_FUNC(seq)(proc, proc_op, &e_ptr[i]);                  \
+			    if (unlikely(rc)) {                                                    \
+				    if (DECODING(proc_op))                                         \
+					    D_FREE(e_ptr);                                         \
+				    goto out;                                                      \
+			    }                                                                      \
+		    }                                                                              \
+		    if (FREEING(proc_op))                                                          \
+			    D_FREE(e_ptr);                                                         \
+	    } next_field_##r:,                                                                     \
+	    BOOST_PP_IF(BOOST_PP_EQUAL(CRT_RAW, CRT_GEN_GET_KIND(seq)),                            \
+			rc = crt_proc_memcpy(proc, proc_op, &ptr->CRT_GEN_GET_NAME(seq),           \
+					     sizeof(CRT_GEN_GET_TYPE(seq)));                       \
+			if (unlikely(rc)) goto out;                                                \
+			, rc = CRT_GEN_GET_FUNC(seq)(proc, proc_op, &ptr->CRT_GEN_GET_NAME(seq));  \
+			if (unlikely(rc)) goto out;))
+
+#define CRT_GEN_PROC_FUNC(type_name, seq)                                                          \
+	static int crt_proc_##type_name(crt_proc_t proc, struct type_name *ptr)                    \
+	{                                                                                          \
+		crt_proc_op_t proc_op;                                                             \
+		int           rc = -DER_INVAL;                                                     \
+		if (unlikely(proc == NULL || ptr == NULL))                                         \
+			goto out;                                                                  \
+		rc = crt_proc_get_op(proc, &proc_op);                                              \
+		if (unlikely(rc))                                                                  \
+			goto out;                                                                  \
+		BOOST_PP_SEQ_FOR_EACH(CRT_GEN_PROC_FIELD, ptr, seq)                                \
+	out:                                                                                       \
```

Comment:
(style)  spaces required around that ':' (ctx:VxE)
(style)  labels should not be indented